File: examples/quickstart.py
"""Minimal Python quickstart for Tokenfold's bytes-first v2 interface."""

import json
from pathlib import Path

import tokenfold

HERE = Path(__file__).parent


def show(label: str, report: tokenfold.CompressionReport) -> None:
    print(f"== {label} ==")
    print(
        f"tokens: {report.original_tokens} -> {report.compressed_tokens} "
        f"({report.saved_tokens} saved, {report.savings_pct:.1f}%)"
    )


body = (HERE / "openai_payload.json").read_bytes()
result = tokenfold.compress(
    body,
    format=tokenfold.InputFormat.OPENAI_JSON,
    preset=tokenfold.Preset.BALANCED,
)
show("OpenAI request", result.report)
print(f"payload: {len(result.payload)} bytes\n")

data = (HERE / "api_response.json").read_bytes()
receipt = tokenfold.inspect(data, format=tokenfold.InputFormat.JSON)
show("generic JSON preview", receipt)

toon_input = json.dumps({"users": [{"id": 1}, {"id": 2}]}).encode()
encoded = tokenfold.compress(toon_input, format="json", encoding=tokenfold.Encoding.TOON)
restored = tokenfold.decode(encoded.payload, from_format="toon")
assert json.loads(restored) == json.loads(toon_input)
show("explicit TOON", encoded.report)

File: examples/quickstart.mjs
/** Minimal Node.js quickstart for Tokenfold's bytes-first v2 interface. */
import { readFile } from "node:fs/promises";

const { compress, decode, inspect } = await import("tokenfold").catch(() =>
  import(new URL("../packages/tokenfold/dist/index.js", import.meta.url).href),
);
const here = (name) => new URL(name, import.meta.url);
const show = (label, report) => console.log(
  `== ${label} ==\ntokens: ${report.original_tokens} -> ${report.compressed_tokens} ` +
  `(${report.saved_tokens} saved, ${report.savings_pct.toFixed(1)}%)`,
);

const body = await readFile(here("openai_payload.json"));
const request = await compress(body, { format: "openai", preset: "balanced" });
show("OpenAI request", request.report);
console.log(`payload: ${request.payload.length} bytes\n`);

const data = await readFile(here("api_response.json"));
show("generic JSON preview", await inspect(data, { format: "json" }));

const toonInput = Buffer.from(JSON.stringify({ users: [{ id: 1 }, { id: 2 }] }));
const encoded = await compress(toonInput, { format: "json", encoding: "toon" });
const restored = await decode(encoded.payload, { from: "toon" });
if (JSON.stringify(JSON.parse(Buffer.from(restored))) !== JSON.stringify(JSON.parse(toonInput))) {
  throw new Error("TOON round trip changed the JSON value");
}
show("explicit TOON", encoded.report);
